### PR TITLE
simpler divergence handling and fix vector data

### LINKED ISSUE #100: Possible error in the l2 loss?
This works:

```julia
using DifferentialEquations
using JuMP
using NLopt
using Plots

#some parameters for DiffEq
const op = (dt       = 0.0001,
            reltol   = 1e-20,
            abstol   = 1e-8,
            maxiters = 10000)
g = @ode_def LorenzExample begin
  dx = σ*(y-x)
  dy = x*(ρ-z) - y
  dz = x*y - β*z
end σ ρ β
u0 = [1.0;0.0;0.0]
tspan = (0.0,1.0)
model_ode(p_) = ODEProblem(g, u0, tspan,p_)
solve_model(mp_) = DifferentialEquations.solve(model_ode(mp_), Tsit5(); op...)
mock_data = getindex.(solve_model([10.0,28.0,8/3]).(0.0:0.01:1.0),1)
# Optimization stuff
function loss_func(sol, data)
    tot_loss = 0.0
    if any((s.retcode != :Success for s in sol))
        tot_loss = Inf
    else
        sdat = getindex.(sol.(0.0:0.01:1.0), 1)
        tot_loss = sum((sdat .- data) .^2)
    end
    return tot_loss
end

loss_objective(mp_, dat) = build_loss_objective(model_ode(mp_), Tsit5(), x-> loss_func(x, dat); op...)

juobj(args...) = loss_objective(args, mock_data)(args)
jumodel = Model()
JuMP.register(jumodel, :juobj, 3, juobj, autodiff=true)
@variables jumodel begin
    σ,(start=8)
    ρ,(start=25.0)
    β,(start=10/3)
end
@NLobjective(jumodel, Min, juobj(σ, ρ, β))
setsolver(jumodel, NLoptSolver(algorithm=:LD_MMA))
sol = JuMP.solve(jumodel)
best_mp = getvalue.(getindex.((jumodel,), Symbol.(jumodel.colNames)))
sol = DifferentialEquations.solve(best_mp |> model_ode, Tsit5(); op...)
plot(getindex.(sol.(0.0:0.01:1.0),1))
scatter!(mock_data, markersize=2)
```

This diverges:

```julia
loss_objective(mp_, dat) = build_loss_objective(model_ode(mp_), Tsit5(), L2Loss(t,dat); op...)

juobj(args...) = loss_objective(args, mock_data)(args)
jumodel = Model()
JuMP.register(jumodel, :juobj, 3, juobj, autodiff=true)
@variables jumodel begin
    σ,(start=8)
    ρ,(start=25.0)
    β,(start=10/3)
end
@NLobjective(jumodel, Min, juobj(σ, ρ, β))
setsolver(jumodel, NLoptSolver(algorithm=:LD_MMA))
sol = JuMP.solve(jumodel)
best_mp = getvalue.(getindex.((jumodel,), Symbol.(jumodel.colNames)))
sol = DifferentialEquations.solve(best_mp |> model_ode, Tsit5(); op...)
plot(getindex.(sol.(0.0:0.01:1.0),1))
scatter!(mock_data, markersize=2)
```

I think it highlights an issue with our L2 loss. Instead of increasing the vector size and such, I think we should just do this strategy of outputting infinite loss when not successful.

Good intro GSoC task. 

@Vaibhavdixit02

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes #100 

```julia
using OrdinaryDiffEq, ParameterizedFunctions, DiffEqParamEstim
using JuMP
using NLopt
using Plots

#some parameters for DiffEq
const op = (dt       = 0.0001,
            reltol   = 1e-20,
            abstol   = 1e-8,
            maxiters = 10000,
            save_idxs = 1,
            saveat = 0.01)
g = @ode_def LorenzExample begin
  dx = σ*(y-x)
  dy = x*(ρ-z) - y
  dz = x*y - β*z
end σ ρ β
u0 = [1.0;0.0;0.0]
t = 0.0:0.01:1.0
tspan = (0.0,1.0)
model_ode(p_) = ODEProblem(g, u0, tspan,p_)
solve_model(mp_) = OrdinaryDiffEq.solve(model_ode(mp_), Tsit5(); op...)
mock_data = solve_model([10.0,28.0,8/3]).u

loss_objective(mp_, dat) = build_loss_objective(model_ode(mp_), Tsit5(), L2Loss(t,dat); op...)

juobj(args...) = loss_objective(args, mock_data)(args)
jumodel = Model()
JuMP.register(jumodel, :juobj, 3, juobj, autodiff=true)
@variables jumodel begin
    σ,(start=8)
    ρ,(start=25.0)
    β,(start=10/3)
end
@NLobjective(jumodel, Min, juobj(σ, ρ, β))
setsolver(jumodel, NLoptSolver(algorithm=:LD_MMA))

juobj([8.0,25.0,10/3]...)

sol = JuMP.solve(jumodel)
best_mp = getvalue.(getindex.((jumodel,), Symbol.(jumodel.colNames)))
sol = OrdinaryDiffEq.solve(best_mp |> model_ode, Tsit5(); op...)
plot(getindex.(sol.(t),1))
scatter!(mock_data, markersize=2)
```

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@Vaibhavdixit02
